### DISCUSSION TITLE
引入翻译专属词库（专业词库、行业词库）功能

### DISCUSSION THREAD

**Discussion by Geo123abc** [immersive-translate/immersive-translate#606] [Ideas]
### 新功能 | New Feature

1. 类似 [趣卡翻译中如何创建专属的翻译词库呢？ - 趣卡词库](  https://xuexi.qukaa.com/publish/1.html  )    ，引入翻译专属词库功能；
2. 这个专属词库可以导入、导出为csv文件格式，方便用户手动维护
3. 专属词库可以设置多个，每个专属词库可以选择性开、关

### 为什么您想要添加这些功能 | Why you want to add these features

1. 可以应对不同的使用场景，增加翻译结果的可读性和流畅性，比如，某些名词在生活语境中翻译肯定和专业文献语境中翻译是不一样的（计算机领域应该有很多这样的词汇）

### 提议 | Proposal

1. 可以参考一下 [趣卡翻译中如何创建专属的翻译词库呢？ - 趣卡词库](  https://xuexi.qukaa.com/publish/1.html  )  ，但是可能有难度；
2. 至于专业词库来源，可以考虑设置一个用户交流区，便于同行之间交流，另外，针对某些行业的mdx格式词典也可以作为很好的专业词库来源；

**Comment by theowenyoung** [comment]
目前沉浸式翻译用的是第三方翻译服务，不太好做集成，词库的替换最好是在翻译服务后端发生，沉浸式翻译这一侧做的话，效果不太好，不过我们也在研究看看有没有其他更好的办法。

**Reply by Geo123abc** [reply]
有个思路，可以试一下，看看效果，以英译中为例：
英文原文 -- 使用专业词库匹配、替换为中文，获得中英混杂的内容 -- 发给 AI 翻译

这个思路对于GPT类的翻译引擎应该效果还可以，对于谷歌翻译这种传统的可能不太好
PS：个人认为，在GPT类大模型看来，英语也好，中文也罢，他们都是一样的，基本上不需要思维转换，需要思维转换的是人，是人把语言分为了中、英等。所以大模型在翻译中英混杂的文字时，是要比传统机器翻译有优势的。另外，专业词库一般匹配的是名词，这应该也有利于大模型翻译中英混合的内容。

**Reply by theowenyoung** [reply]
是的，如果是大模型的话，我们甚至直接在提示词里 加入本段相关的专业术语，告诉大模型这些术语要这样翻译，应该也能得到不错的结果，但是这个就只适合大模型，然后也会让翻译工具使用变得更复杂。我们研究一下。
